cpp/ql/src/semmle/code/cpp/PrintAST.qll

```diff
@@ -341,6 +341,20 @@ class CastNode extends ConversionNode {
   }
 }
 
+/**
+ * A node representing a `StmtExpr`.
+ */
+class StmtExprNode extends ExprNode {
+  StmtExpr stmtExpr;
+
+  StmtExprNode() { stmtExpr = expr }
+
+  override ASTNode getChildInternal(int childIndex) {
+    childIndex = 0 and
+    result.getAST() = stmtExpr.getStmt()
+  }
+}
```

Comment:
```suggestion
/**
 * A node representing a `StmtExpr`.
 */
class StmtExprNode extends ExprNode {
  override StmtExpr expr;

  override ASTNode getChildInternal(int childIndex) {
    childIndex = 0 and
    result.getAST() = expr.getStmt()
  }
}
```
The rest of `PrintAST.qll` doesn't (yet) follow this pattern, but you can just override the field from the super class with a more specific type, and leave out the charpred.